include/onnxruntime/core/framework/allocator.h

```diff
@@ -15,21 +15,80 @@
 #include "core/framework/fence.h"
 #include "core/session/onnxruntime_c_api.h"
 
+// Struct to represent a physical device.
+struct OrtDevice {
+  using DeviceType = int8_t;
```

Comment:
Please use uint8_t, unless you plan to allow negative values for DeviceType. Not sure why negative values would make sense for device type. Ditto for MemoryType and DeviceId.